BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -0,0 +1,400 @@
+            
+        case SBAProfileTypeIdentifier.bool:
+            return newValue as? NSNumber != nil
+            
+        case SBAProfileTypeIdentifier.date:
+            return newValue as? NSDate  != nil
+            
+        case SBAProfileTypeIdentifier.hkBiologicalSex:
+            return newValue as? HKBiologicalSex != nil
+            
+        case SBAProfileTypeIdentifier.hkQuantity:
+            return newValue as? NSNumber != nil
+            
+        default:
+            return false
+        }
+    }
+}
+
+class SBAProfileItemBase: NSObject, SBAProfileItem {
+    /**
+     The value property is used to get and set the profile item's value in whatever internal data
+     storage is used by the implementing class.
+     */
+    open var value: Any?
+
+    fileprivate let sourceDict: [AnyHashable: Any]
+    
+    open var profileKey: String {
+        get {
+            return sourceDict["profileKey"]! as! String
+        }
+    }
+    
+    open var sourceKey: String {
+        get {
+            return sourceDict["sourceKey"] as! String? ?? self.profileKey
+        }
+    }
+    
+    open var demographicKey: String {
+        get {
+            return sourceDict["demographicKey"] as! String? ?? self.profileKey
+        }
+    }
+    
+    open var itemType: SBAProfileTypeIdentifier {
+        get {
+            return sourceDict["itemType"]! as! SBAProfileTypeIdentifier
+        }
+    }
+    
+    public required init(dictionaryRepresentation dictionary: [AnyHashable: Any]) {
+        sourceDict = dictionary
+        super.init()
+    }
+    
+    open var jsonValue: SBBJSONValue? {
+        get {
+            return self.commonJsonValueGetter()
+        }
+        
+        set {
+            commonJsonValueSetter(value: newValue)
+        }
+    }
+    
+    open var demographicJsonValue: SBBJSONValue? {
+        get {
+            return self.commonDemographicJsonValue()
+        }
+    }
+}
+
+class SBAKeychainProfileItem: SBAProfileItemBase {
+    private var keychain: SBAKeychainWrapper
+    
+    public required init(dictionaryRepresentation dictionary: [AnyHashable: Any]) {
+        keychain = SBAUser.shared.keychain
+        
+        let keychainService = dictionary["keychainService"] as! String?
+        let keychainAccessGroup = dictionary["keychainAccessGroup"] as! String?
+        
+        if (keychainService != nil || keychainAccessGroup != nil) {
+            keychain = SBAKeychainWrapper(service: keychainService, accessGroup: keychainAccessGroup)
+        }
+        super.init(dictionaryRepresentation: dictionary)
+    }
+    
+    override open var value: Any? {
+        get {
+            var err: NSError?
+            let obj = keychain.object(forKey: profileKey, error: &err)
+            if let error = err {
+                print("Error accessing keychain \(profileKey): \(error.code) \(error)")
+            }
+            return self.typedValue(from: obj)
```

Comment:
This implementation of typedValue will just return nil. Subclasses might override to do something else.